### ISSUE TITLE
Overly complicated system

### ISSUE THREAD

**Issue by shmezi** [Auties00/Cobalt#399]
Hey, I've tried using this library multiple times over the past year or so.. Every time I always run into two issues, the api constantly changes to the point where I have to rewrite and figure out how to do stuff over again from version to version.
And this library is WAY too complicated to actually use for anything, too many options in the first readme alone.
IMO this is a great project that needs some actual documentation and the main structure cant change every too seconds.

**Comment by Auties00** [comment]
> Hey, I've tried using this library multiple times over the past year or so.. Every time I always run into two issues, the api constantly changes to the point where I have to rewrite and figure out how to do stuff over again from version to version.
> And this library is WAY too complicated to actually use for anything, too many options in the first readme alone.
> IMO this is a great project that needs some actual documentation and the main structure cant change every too seconds.

I agree with you on the frequent changes, I just love to code so I frequently switch things up. That's why I've changed the project's artifact to cobalt and I will commit to no breaking changes starting from release 0.1 
On the other hand I'd say that the documentation is excellent, and that the complexity is justified because there just are a lot of things you can do with this library. If you are not familiar with asynchronous programming in Java and modern features like lambdas I can see how it could be overwhelming, but that's not an issue of the library.

**Comment by shmezi** [comment]
It's so overly complicated I can never tell if theres a bug or if I messed something up..
like some listeners plain out right dont work and I dont know if its me that is using them wrongly or its just broken.
also the examples are outdated.

**Comment by Auties00** [comment]
> It's so overly complicated I can never tell if theres a bug or if I messed something up..
> like some listeners plain out right dont work and I dont know if its me that is using them wrongly or its just broken.
> also the examples are outdated.

As I said I don't feel like the library is hard to use, about your question realistically it's a bug as the master branch has been updated for a long time without a release. When I release 0.1 these issues should be resolved. About the outdated examples, I'll update them when I release 0.1

**Comment by Auties00** [comment]
> Is 0.1 stable and if so is there a way for me to use it

It's on the master branch, but no, it's not stable as of now. You can try it if you want though
